images/git-custom-k8s-auth/Dockerfile

```diff
@@ -12,26 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM google/cloud-sdk:461.0.0 as builder
+FROM google/cloud-sdk:483.0.0 as builder
 # https://github.com/GoogleCloudPlatform/cloud-sdk-docker/blob/master/Dockerfile
 
 ARG AWS_IAM_AUTHENTICATOR_VERSION
+ARG AZURE_KUBELOGIN_VERSION
+
 RUN curl -fsSL \
     https://github.com/kubernetes-sigs/aws-iam-authenticator/releases/download/v${AWS_IAM_AUTHENTICATOR_VERSION}/aws-iam-authenticator_${AWS_IAM_AUTHENTICATOR_VERSION}_linux_amd64 -o /aws-iam-authenticator && \
     chmod +x /aws-iam-authenticator
 
+RUN apt update && apt install -y unzip && curl -fsSLO https://github.com/Azure/kubelogin/releases/download/v${AZURE_KUBELOGIN_VERSION}/kubelogin-linux-amd64.zip \
+ && unzip -j kubelogin-linux-amd64.zip -d / \
+ && rm kubelogin-linux-amd64.zip
 
 # Update the date below (check https://hub.docker.com/_/alpine/tags) to get the
 # alpine edge base image, which has a faster release cadence than alpine:latest
 # (which tracks alpine stable), to pick up security patches more quickly.
-FROM alpine:20231219
+FROM alpine:20240606
```

Comment:
TODO: switch this, cc @dims 

I would recommend something like how we did distroless-iptables

Reply:
what's the issue with alpine ?

Reply:
@ameukam https://github.com/kubernetes/enhancements/blob/master/keps/sig-release/1729-rebase-images-to-distroless/README.md#type-3-alpine-based-images

Reply:
let's add a TODO and/or an issue to switch this off of alpine for later

Reply:
(also we have DNS resolver flakes when using a less mature libc)

Reply:
TODO added